src/dune_rules/pkg_rules.ml

```diff
@@ -1175,13 +1189,40 @@ end = struct
           (Context_name.build_dir ctx)
           (Dune_pkg.Lock_dir.Pkg.files_dir info.name ~lock_dir)
       in
+      let* inherited_post_depends =
+        (* Deduplicate package names, preserving the association
+           between package names and (any) one of the locations of the
+           package name (for error messages). *)
+        let all_locs_by_name =
+          List.concat_map depends ~f:(fun (pkg : Pkg.t) ->
+            List.map pkg.post_depend_names ~f:(fun (loc, name) -> name, loc))
+          |> Package.Name.Map.of_list_reduce ~f:(fun x _ -> x)
```

Comment:
This step does re-ordering of deps

Reply:
Ah good point. Will fix.

Reply:
Ok I changed this to preserve the order in which post dependencies appear in the lockfile.